btree/btree.c

```diff
@@ -6362,13 +7282,216 @@ static void vkvv_capture(struct slot *slot, int cr, struct m0_be_tx *tx)
+            M0_ASSERT(filled < max);
+        }
+    }
+    m0_free(queue);
+    return buf;
+}
+
+M0_UNUSED static void check_dbg_tree(struct td *tree)
+{
+	struct nd *root = tree->t_root;
+	struct nd **queue = m0_alloc(1000 * sizeof(struct nd *));
+	int front = 0, rear = 0;
+	struct slot prev;
+	struct slot next;
+	struct nd *prev_node;
+	int prev_idx;
+	bool is_first = true;
+
+	void                    *p_key_prev;
+	m0_bcount_t              ksize_prev;
+	void                    *p_key_next;
+	m0_bcount_t              ksize_next;
+	prev.s_rec.r_key.k_data = M0_BUFVEC_INIT_BUF(&p_key_prev, &ksize_prev);
+	next.s_rec.r_key.k_data = M0_BUFVEC_INIT_BUF(&p_key_next, &ksize_next);
+
+    queue[front] = root;
+    while (front != -1 && rear != -1) {
+	//pop one elemet
+	struct nd* node = queue[front];
+	int level;
+	if (front == rear) {
+	    front = -1;
+	    rear = -1;
+	} else
+	    front++;
+	level = bnode_level(node);
+	if (level > 0) {
+	    int total_count = bnode_count(node);
+	    int j;
+	    struct segaddr child_node_addr;
+	    struct slot    node_slot = {};
+	    struct node_op  i_nop;
+	    for(j=0 ; j < total_count; j++) {
+	        struct segaddr child_addr;
+	        struct slot    nd_slot = {};
+	        struct node_op  ic_nop = {};
+	        nd_slot.s_node = node;
+	        nd_slot.s_idx = j;
+	        bnode_child(&nd_slot, &child_addr);
+	        ic_nop.no_opc = NOP_LOAD;
+	        bnode_get(&ic_nop, tree, &child_addr, P_NEXTDOWN);
+	        if (front == -1) {
+	            front = 0;
+	        }
+	        rear++;
+	        if (rear == 9999) {
+	            // printf("***********OVERFLW***********");
+	            M0_LOG(M0_ALWAYS,"Overflow !!!!!!");
+	            break;
+	        }
+	        queue[rear] = ic_nop.no_node;
+	    }
+	    //store last child:
+	    node_slot.s_node = node;
+	    node_slot.s_idx = j;
+	    bnode_child(&node_slot, &child_node_addr);
+	    i_nop.no_opc = NOP_LOAD;
+	    bnode_get(&i_nop, tree, &child_node_addr, P_NEXTDOWN);
+	    if (front == -1) {
+	        front = 0;
+	    }
+	    rear++;
+	    if (rear == 9999) {
+	        // printf("***********OVERFLW***********");
+	        M0_LOG(M0_ALWAYS,"Overflow !!!!!!");
+	        break;
+	    }
+	    queue[rear] = i_nop.no_node;
+
+	} else {
+	    int j;
+	    int total_count = bnode_count(node);
+	    for(j = 0 ; j < total_count; j++) {
+		next.s_node = node;
+		next.s_idx = j;
+		bnode_key(&next);
+		if (!is_first) {
+			prev.s_node = prev_node;
+			prev.s_idx = prev_idx;
+			bnode_key(&prev);
+
+
+			if (vkvv_cmp_keys(prev.s_rec.r_key, next.s_rec.r_key) == false)
+			{
+				char *ln = NULL;
+				#ifndef __KERNEL__
+				ln = malloc(1048576);
+				#endif
+				if (ln != NULL)
+					memset(ln, 0, 1048576);
+
+				M0_LOG(M0_ALWAYS,"PUT T(t=%p) >%s", tree,
+				vkvv_dbg_tree(tree, ln, 1048576));
+				M0_ASSERT(0);
+			}
+		} else
+			is_first = false;
+		prev_node = node;
+		prev_idx = j;
+	    }
+	}
+    }
+    m0_free(queue);
+}
+
```

Comment:
These functions are not called from anywhere. Please remove them.

Reply:
Done